extension-workspace-reference.md

```diff
@@ -0,0 +1,121 @@
+}
+```
+
+_Response_:
+* result: `ReferenceInformation[]` defined as follows:
+```typescript
+/**
+ * Represents information about a reference to programming constructs like
+ * variables, classes, interfaces etc.
+ */
+interface ReferenceInformation {
+	/**
+	 * The location in the workspace where the `symbol` has been referenced.
+	 */
+	reference: Location;
+
+	/**
+	 * Metadata information describing the symbol being referenced.
+	 */
+	symbol: ReferenceSymbolInformation;
+}
+```
+* error: code and message set in case an exception happens during the workspace reference request.
+
+Where `ReferenceSymbolInformation` is defined as follows:
+
+```typescript
+/**
+ * Represents information about a programming construct like a variable, class,
+ * interface etc that has a reference to it. Effectively, it contains data similar
+ * to SymbolInformation except all fields are optional and a metadata field is
+ * present for language-specific data.
+ *
+ * ReferenceSymbolInformation does NOT always uniquely identify a symbol. The
+ * caller should effectively consider the returned information to be
+ * information about the symbol which _generally_ (but NOT always) identifies a
+ * single symbol.
+ */
+interface ReferenceSymbolInformation {
+    /**
+     * The name of this symbol (same as `SymbolInformation.name`).
+     */
+    name?: string;
+
+    /**
+     * The kind of this symbol (same as `SymbolInformation.kind`).
+     */
+    kind?: number;
+
+    /**
+     * The file URI of this symbol (same as `SymbolInformation.location.uri`).
+     */
+    file?: string;
+
+    /**
+     * The name of the symbol containing this symbol (same as `SymbolInformation.containerName`).
+     */
+    containerName?: string;
+
+    /**
+     * The repository URI that the symbol is defined in, e.g. `github.com/golang/go`
+     * or `bitbucket.org/jespern/django-piston`.
+     */
+    repo?: string;
+
+    /**
+     * The 'package', 'library', or 'crate' name that the symbol is defined in.
+     * For example, in JS/TS this would be the npm module name. In Go, the full
+     * package import path. In PHP, the Composer package name. etc.
+     */
+    packageName?: string;
+
+    /**
+     * Metadata describing the symbol that is being referenced. It is up to the
+     * language server to define what exact data this object contains.
+     *
+     * A language server is encouraged to include additional information about
+     * the symbol. Specifically, any information that a user querying against
+     * the returned information might be interested in. For example, with Go, a
+     * user may wish to include or exclude vendored packages from their search
+     * query, so the Go language server would include `"vendor": true` or `"vendor": false`
+     * entry in the response which someone indexing this object could filter
+     * based on. Any information not already covered by the optional standardized
+     * fields above.
+     */
+    meta: Object;
```

Comment:
Tbh I think we should remove `meta`. Instead, if we have a use case for a property `vendor`, we should add it as an optional property and every language that has a similar concept can use it. Otherwise we will end up with custom properties for each language that mean the same thing, which is the whole problem LSP tries to solve.